src/sql/base/common/dataProvider.ts

```diff
@@ -0,0 +1,17 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Event } from 'vs/base/common/event';
+
+export interface IDisposableDataProvider<T> extends Slick.DataProvider<T> {
```

Comment:
It'd be nice to add descriptions to all these properties while you're modifying this stuff - they're not all exactly clear on what they're doing exactly